packages/bsky/src/services/moderation/status.ts

```diff
@@ -0,0 +1,272 @@
+// This may require better organization but for now, just dumping functions here containing DB queries for moderation status
+
+import { AtUri } from '@atproto/syntax'
+import { PrimaryDatabase } from '../../db'
+import {
+  ModerationEvent,
+  ModerationSubjectStatus,
+} from '../../db/tables/moderation'
+import {
+  ACKNOWLEDGE,
+  REVIEWOPEN,
+  MUTE,
+  REVIEWCLOSED,
+  REPORT,
+  REVIEWESCALATED,
+  REVERT,
+  TAKEDOWN,
+  ESCALATE,
+} from '../../lexicon/types/com/atproto/admin/defs'
+import { ModerationEventRow, ModerationSubjectStatusRow } from './types'
+
+const actionTypesImpactingStatus = [
+  ACKNOWLEDGE,
+  REPORT,
+  ESCALATE,
+  REVERT,
+  TAKEDOWN,
+  MUTE,
+]
+
+// TODO: How do we handle revert? for "revert" event we will have a reference event id that is being reversed
+// We will probably need a helper that can take a list of events and compute the final state of the subject
+// That helper will have to be invoked here with all events up until the point where the reverted event was created
+const getSubjectStatusForModerationEvent = ({
+  action,
+  durationInHours,
+}: {
+  action: string
+  durationInHours: number | null
+}): Partial<ModerationSubjectStatusRow> | null => {
+  switch (action) {
+    case ACKNOWLEDGE:
+      return {
+        reviewState: REVIEWCLOSED,
+        lastReviewedAt: new Date().toISOString(),
+      }
+    case REPORT:
+      return {
+        reviewState: REVIEWOPEN,
+        lastReportedAt: new Date().toISOString(),
+      }
+    case ESCALATE:
+      return {
+        reviewState: REVIEWESCALATED,
+        lastReviewedAt: new Date().toISOString(),
+      }
+    // REVERT can only come through when a revert event is emitted but there are no status impacting event
+    // before it. In which case, we will default to REVIEWCLOSED
+    case REVERT:
+      return {
+        reviewState: REVIEWCLOSED,
+        lastReviewedAt: new Date().toISOString(),
+      }
+    case TAKEDOWN:
+      return {
+        takendown: true,
+        reviewState: REVIEWCLOSED,
+        lastReviewedAt: new Date().toISOString(),
+      }
+    case MUTE:
+      return {
+        // By default, mute for 24hrs
+        muteUntil: new Date(
+          Date.now() + (durationInHours || 24) * 60 * 60 * 1000,
```

Comment:
We should have some constants kicking around in the @atproto/common package `DAY`/`HOUR` for the number of milliseconds in a day/hour.